src/viam/sdk/components/private/arm_server.cpp

```diff
@@ -53,6 +53,32 @@ ::grpc::Status ArmServer::MoveToJointPositions(
     });
 }
 
+::grpc::Status ArmServer::MoveThroughJointPositions(
+    ::grpc::ServerContext*,
+    const ::viam::component::arm::v1::MoveThroughJointPositionsRequest* request,
+    ::viam::component::arm::v1::MoveThroughJointPositionsResponse*) noexcept {
+    return make_service_helper<Arm>(
+        "ArmServer::MoveThroughJointPositions", this, request)([&](auto& helper, auto& arm) {
+        std::vector<std::vector<double>> positions;
+
+        positions.reserve(request->positions_size());
+        for (const auto& values : request->positions()) {
+            positions.push_back({values.values().begin(), values.values().end()});
```

Comment:
This can possibly be written `positions.emplace_back(values.values().begin(), values.values().end)` which will avoid a move and construct the vector in place.